src/Platform/Vulkan/VulkanPipeline.cc

```diff
@@ -0,0 +1,44 @@
+#include "VulkanPipeline.hh"
+
+// libs
+#include "volk.h"
+
+// Platform
+#include "VulkanDevice.hh"
+#include "VulkanFrameManager.hh"
+
+namespace esp
+{
+  VulkanPipeline::VulkanPipeline(VkPipelineLayout pipeline_layout,
+                                 VkPipeline graphics_pipeline,
+                                 std::unique_ptr<VulkanUniformManager> uniform_manager) :
+      m_pipeline_layout(pipeline_layout),
+      m_graphics_pipeline(graphics_pipeline), m_uniform_manager(std::move(uniform_manager))
+  {
+  }
+
+  VulkanPipeline::~VulkanPipeline()
+  {
+    vkDestroyPipelineLayout(VulkanDevice::get_logical_device(), m_pipeline_layout, nullptr);
+    vkDestroyPipeline(VulkanDevice::get_logical_device(), m_graphics_pipeline, nullptr);
+  }
+
+  void VulkanPipeline::attach_uniforms() const { m_uniform_manager->attach_ds(m_pipeline_layout); }
```

Comment:
@DawidSzpejna I don't really like using such shortcuts, because right now I have to figure out what the hell is "ds" supposed to be. If this method was called "attach_descriptor_set" there wouldn't be a problem.